xdsl/transforms/reconcile_unrealized_casts.py

```diff
@@ -89,27 +94,32 @@ def gen_all_uses_cast(
         rewriter.erase_op(cast)
 
 
+@dataclass
 class ReconcileUnrealizedCastsPattern(RewritePattern):
     """
     Removes the chains of `builtin.unrealized_conversion_cast` operations
     that are no longer necessary and that start with the matched
     `builtin.unrealized_conversion_cast`.
     """
 
+    with_warnings: bool = False
+
     @op_type_rewrite_pattern
     def match_and_rewrite(
         self, op: builtin.UnrealizedConversionCastOp, rewriter: PatternRewriter
     ):
-        _try_remove_cast_chain(op, rewriter)
+        _try_remove_cast_chain(op, rewriter, self.with_warnings)
 
 
-def reconcile_unrealized_casts(module: ModuleOp):
+def reconcile_unrealized_casts(module: ModuleOp, with_warnings: bool = True):
```

Comment:
I'd suggest a couple of small changes here and in the pattern:
```suggestion
def reconcile_unrealized_casts(module: ModuleOp, *, warn_on_error: bool = True):
```
The first is to make the flag a keyword argument. Someone reading code that calls this functions will need to look at the implementation of the function to understand what `reconcile_unrealized_casts(op, True)` means, whereas `reconcile_unrealized_casts(op, warn_on_error=True)` is self-documenting. I'd also say that my suggested renaming is a little clearer about the effect of the flag, although that's a nit/subjective thing, I think there are a bunch of ways to phrase it, and the one in this PR is ok also.